src/test/java/org/folio/moduserstest/ProxyRelationshipsIT.java

```diff
@@ -0,0 +1,278 @@
+
+import io.vertx.core.Vertx;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import lombok.SneakyThrows;
+
+/**
+ * At the moment, proxy relationships do not require the user to exist
+ * this means that these tests do not currently create users for any of the
+ * users involved.
+ * This is NOT how the system is intended to be used
+ */
+@ExtendWith(VertxExtension.class)
+@Timeout(value = 20, unit = SECONDS)
+class ProxyRelationshipsIT {
+  private static ProxiesClient proxiesClient;
+
+  @BeforeAll
+  @SneakyThrows
+  public static void beforeAll(Vertx vertx, VertxTestContext context) {
+    PostgresClient.setPostgresTester(new PostgresTesterContainer());
+
+    int port = NetworkUtils.nextFreePort();
+
+    final var headers = new OkapiHeaders("http://localhost:" + port,
+      "diku", "diku");
+
+    proxiesClient = new ProxiesClient(new URI("http://localhost:" + port), headers);
+
+    final var module = new VertxModule(vertx);
+
+    module.deployModule(port)
+      .onComplete(context.succeeding(res -> module.enableModule(headers,
+          false, false)
+        .onComplete(context.succeedingThenComplete())));
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    proxiesClient.deleteAllProxies();
+  }
+
+  @Test
+  void userMayNominateAProxy() {
+    final var userId = generateId();
+    final var proxyUserId = generateId();
+
+    final var relationship = proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(proxyUserId)
+        .build());
+
+    assertThat(relationship.getId(), is(notNullValue()));
+    assertThat(relationship.getUserId(), is(userId));
+    assertThat(relationship.getProxyUserId(), is(proxyUserId));
+  }
+
+  @Test
+  void canGetProxyRelationship() {
+    final var userId = generateId();
+    final var proxyUserId = generateId();
+
+    final var createdRelationship = proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(proxyUserId)
+        .build());
+
+    final var fetchedRelationship = proxiesClient.getProxyRelationship(
+      createdRelationship.getId());
+
+    assertThat(fetchedRelationship.getId(), is(createdRelationship.getId()));
+    assertThat(fetchedRelationship.getUserId(), is(userId));
+    assertThat(fetchedRelationship.getProxyUserId(), is(proxyUserId));
+  }
+
+  @Test
+  void userMayHaveMultipleProxies() {
+    final var userId = generateId();
+    final var firstProxyUserId = generateId();
+    final var secondProxyUserId = generateId();
+
+    proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(firstProxyUserId)
+        .build());
+
+    proxiesClient.attemptToCreateProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(secondProxyUserId)
+        .build())
+      .statusCode(is(HTTP_CREATED));
```

Comment:
Either add more tests or use the `.createProxyRelationship` that has built-in statusCode test.

Reply:
What additional tests would you want here?

I chose to attempt and an explicit status check in the test because it makes the interesting expectations more obvious.

I found that using the create variation of the method was somewhat ambiguous about it's intent and that is better used for intermediary steps where a specific response is needed yet not an important aspect of the test.

This was a realisation I had during the conversion of these tests and so some of the others use the regular create method which is an inconsistency that I could address.